### ISSUE TITLE
[Grid] Multiselect in Grid

### ISSUE THREAD

**Issue by klaasjs** [sl-design-system/components#3228]
### Provide a general summary of the feature here

Right now one can select multiple items by clicking the checkbox.However if you have say 20 rows and you want to select 13 of them then you need 8 clicks. The select all and then the deselect of 7 items.

In Windows Explorer / Finder it is possible to use the cmd key to select multiple individual items and the shift key to select multiple continuous items. Maybe this would be nice to have in sl-grid?

### 🤔 Expected Behavior?

In a grid with 10 rows. I click the first row, I click the 5th row with shift key pressed: all the rows from 1 until 5 are selected.

In a grid with 10 rows. I click the first row, I click somewhere in the 4th row (not necessarily on the checkbox) the 4th row is selected.

### 😯 Current Behavior

both shift and ctrl/cmd keys are not used.

### 💁 Possible Solution

That it selects like Windows Explorer / Mac Finder.

### 🔦 Context

I think our users want to perform actions fast. So to give them the possibility to make less clicks means they are faster 'done'. They can drink coffee and feel satisfied with their work faster which i hope is a good thing..

### 👤 Your name

Klaas-Jan Stellema

### 🧢 Your Product/Team

Magister/Us Mem

**Comment by a11ymiko** [comment]
This is a good idea, but it has to be tested with screen reader in very early stage of development.
For screen readers like NVDA `Ctrl` and `Shift` are used as function keys and their use in grid for multiple select could interfere with screen reader.

We should also think if we want this feature to only be possible for mouse users or should this work the same for keyboard-only users? If yes, then there are some additional requirements like information about keyboard shortcuts that should be announced to screen reader users.
